src/NLog.Web.AspNetCore/NLogBufferingTargetWrapperMiddleware.cs

```diff
@@ -44,11 +44,12 @@ public async Task Invoke(HttpContext context)
             }
             finally
             {
-                var targets = LogManager.Configuration.AllTargets;
-                Parallel.ForEach(targets, target =>
```

Comment:
Think the overhead from doing `Parallel.ForEach` is larger than just running on the actual thread.

Also by scheduling on a secondary thread, means that HttpContext can become disposed/unavailable, so it is impossible for the secondary thread to access the HttpContext-Items-dictionary (when it is actual executed).

Reply:
Sure, we can just use normal for each.
Should we take all the code out of middleware and http module and just have static AspNetBufferWrapper.FlushAll method?

Reply:
Yes a static flush method on AspNetBufferWrapper would be good (Taking the HttpContext as input-parameter)

Reply:
Completed